Userland/Utilities/gev.cpp

```diff
@@ -0,0 +1,295 @@
+#include <LibCore/MimeData.h>
+#include <LibGUI/Application.h>
+#include <LibGUI/Painter.h>
+#include <LibGUI/Widget.h>
+#include <LibGUI/Window.h>
+#include <unistd.h>
+
+class EventViewerWidget : public GUI::Widget {
+    C_OBJECT(EventViewerWidget);
+
+public:
+    void set_all_enabled(bool all_enabled) { m_all_enabled = all_enabled; }
+    void set_mouse_enabled(bool mouse_enabled) { m_mouse_enabled = mouse_enabled; }
+    void set_key_enabled(bool key_enabled) { m_key_enabled = key_enabled; }
+    void set_drag_enabled(bool drag_enabled) { m_drag_enabled = drag_enabled; }
+    void set_drop_enabled(bool drop_enabled) { m_drop_enabled = drop_enabled; }
+    void set_resize_enabled(bool resize_enabled) { m_resize_enabled = resize_enabled; }
+    void set_enter_enabled(bool enter_enabled) { m_enter_enabled = enter_enabled; }
+    void set_leave_enabled(bool leave_enabled) { m_leave_enabled = leave_enabled; }
+    void set_theme_enabled(bool theme_enabled) { m_theme_enabled = theme_enabled; }
+    void set_fonts_enabled(bool fonts_enabled) { m_fonts_enabled = fonts_enabled; }
+
+protected:
+    virtual ~EventViewerWidget() override
+    {
+    }
+
+    virtual void paint_event(GUI::PaintEvent& event) override
+    {
+        GUI::Painter painter(*this);
+        painter.add_clip_rect(event.rect());
+
+        painter.fill_rect(rect(), Color::White);
+    }
+    virtual void resize_event(GUI::ResizeEvent& event) override
+    {
+        if (!(m_all_enabled || m_resize_enabled))
+            return;
+        outln("Resize: ");
+        outln(String::formatted("    Size: {}", event.size()));
+        outln();
+    }
+    virtual void keydown_event(GUI::KeyEvent& event) override
+    {
+        print_key_event(event, true);
+    }
+    virtual void keyup_event(GUI::KeyEvent& event) override
+    {
+        print_key_event(event, false);
+    }
+    virtual void mousemove_event(GUI::MouseEvent& event) override
+    {
+        print_mouse_event(event, "MouseMove:");
+    }
+    virtual void mousedown_event(GUI::MouseEvent& event) override
+    {
+        print_mouse_event(event, "MouseDown:");
+    }
+    virtual void mouseup_event(GUI::MouseEvent& event) override
+    {
+        print_mouse_event(event, "MouseUp:");
+    }
+    virtual void mousewheel_event(GUI::MouseEvent& event) override
+    {
+        print_mouse_event(event, "MouseWheel:");
+    }
+    virtual void doubleclick_event(GUI::MouseEvent& event) override
+    {
+        print_mouse_event(event, "DoubleClick:");
+    }
+    virtual void context_menu_event(GUI::ContextMenuEvent& event) override
+    {
+        outln("ContextMenu:");
+        outln(String::formatted(
+            "    Position: {}, ScreenPosition: {}",
+            event.position(), event.screen_position()));
+        outln();
+    }
+    virtual void enter_event(Core::Event&) override
+    {
+        if (!(m_all_enabled || m_enter_enabled))
+            return;
+        outln("Enter:\n");
```

Comment:
Please prefer two `outln` calls if you want an additional newline, this just looks like a mistake (printf-like use of an API that adds a newline already)